classes/MockDelegateFunctionBuilder.php

```diff
@@ -56,7 +56,7 @@ public function build($functionName = null)
          * If a class with the same signature exists, it is considered equivalent
          * to the generated class.
          */
-        $hash = md5($signatureParameters);
+        $hash = md5(is_string($signatureParameters) ? $signatureParameters : '');
```

Comment:
actually the issue is somewhere else - `getSignatureParameters` should always return `string` UNLESS `build of the `ParameterBuilder` returned earlier when the function you want to mock does not exist.

I would need to know more about your use case, as the usage seems to be wrong.

We can fix it probably by adding:

```php
if ($signatureParameters === null) {
    return;
}
```

before generating the `hash` but personally I would prefer a fix in `phpmock/phpmock` on `ParameterBuilder` and default value of the property (null => empty string).

Reply:
Just created: https://github.com/php-mock/php-mock/pull/41

Thanks